src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java

```diff
@@ -282,8 +284,33 @@ private static Credentials secretToCredentials(Secret secret) {
                     data.get(OPENSHIFT_SECRETS_DATA_USERNAME),
                     data.get(OPENSHIFT_SECRETS_DATA_SSHPRIVATEKEY));
         default:
-            logger.log(Level.WARNING,
-                    "Unknown secret type: " + secret.getType());
+			String type = secret.getType();
+			if (type == null || type.length() == 0) {
+				// the type field is marked optional in k8s.io/api/core/v1/types.go,
+				// default to OPENSHIFT_SECRETS_DATA_SECRET_TEXT in this case
+				String text = "";
+				if (data != null && data.size() > 0) {
+					// convert to JSON for parsing ease in pipelines
+					try {
+						text = new ObjectMapper().writeValueAsString(data);
```

Comment:
is there a jenkins credential type that allows key/value mapping?  It would seem to be easier to consume this for the end user if we gave them a map of key/values direct from the secret, rather than a json text string they need to re-parse.

Reply:
There is not such a type.  Neither in the plain-credentials jar or upstream jenkins of the credentials.jar from the cloudbees credentials plugins that we include / cite as a dependency.

Reply:
ok. that's too bad.